src/main/java/AddCommand.java

```diff
@@ -0,0 +1,60 @@
+public class AddCommand extends Command {
+
+    private String input;
+    private int counter;
+    private int type;
+
+    AddCommand(Tasks[] tasks, String input, int counter, int type) {
+        super(tasks);
+        this.input = input;
+        this.counter = counter;
+        this.type = type;
+    }
+
+    public Tasks[] addTask() {
+        if (type == 0) {
+            int startIndex = input.indexOf("n");
+            int endIndex = input.indexOf("\\at");
+            String preciseInput = input.substring(startIndex+3,endIndex-1).trim();
+            Deadline temp = new Deadline(preciseInput,input.split(" ")[0],input.substring(endIndex+4));
```

Comment:
Please consider using a better name for intermediate variable instead of "temp". Maybe "temporaryDeadline" or "newDeadline". The name should explain a bit more of what the variable is for.